pybamm/simulation.py

```diff
@@ -83,6 +110,126 @@ def __init__(
 
             warnings.filterwarnings("ignore")
 
+    def set_up_experiment(self, model, experiment):
```

Comment:
I found this pretty confusing to follow. It's really neat, but took a few goes through!

Reply:
it's probably fine, but it could be more explicit that we handle experiments by creating a submodel for each instruction and then appending all the solutions together. I think that is obvious when you look at the big picture (e.g. seeing the solution object), but might not be super clear if you just looked here